schema.md

```diff
@@ -1,82 +1,71 @@
 
-/model
-Each backend can store a representation of the model in here. (I guess
-source code for stan, no clue for pymc3 at this point.)
+Moreover, each group contains the following attributes:
+* `created_at`: the date of creation of the group.
+* `inference_library`: the library used to run the inference.
+* `inference_library_version`: version of the inference library used.
 
-/sample_stats
-Statistics computed while sampling, like step size (step_size), depth, diverging, energy (for HMC), log probability (lp), and log likelihood (log_likelihood).
+`InferenceData` data objects contain any combination the groups described below.
 
-/initial_point
-    One point in parameter space for each chain, where that chain started.
-    TODO We could also store that as an attribute to each var in /trace
+#### `posterior`
+Samples from the posterior distribution p(theta|y).
 
-/tuning_trace?
-    Same as /trace, but optional. It contains the trace during tuning. Optional. (only if store_tune=True or so?)
+#### `sample_stats`
+Information and diagnostics about each `posterior` sample, provided by the inference backend. It may vary depending on the algorithm used by the backend (i.e. an affine invariant sampler has no energy associated). The name convention used for `sample_stats` variables is the following:
+* `lp`: unnormalized log probability of the sample
+* `step_size`
+* `step_size_bar`
+* `tune`: boolean variable indicating if the sampler is tuning or sampling
+* `depth`:
+* `tree_size`:
+* `mean_tree_accept`:
+* `diverging`: HMC-NUTS only, boolean variable indicating divergent transitions
+* `energy`: HMC-NUTS only
+* `energy_error`
+* `max_energy_error`
 
-/tuning_advi?
-    Some data about an initialization advi run? History of convergence stats, final sd and mu params? Optional
+#### `observed_data`
+Observed data on which the `posterior` is conditional. It should only contain data which is modeled as a random variable. Each variable should have a counterpart in `posterior_predictive`. The `posterior_predictive` counterpart variable may have a different name.
 
-/divergences
-    Points in parameter space where the leapfrog starts that lead to a divergence (excluding tuning).
-    Does stan have access to that info? We could also just store the accepted point of a divergent trajectory.
+#### `posterior_predictive`
+Posterior predictive samples p(y|y) corresponding to the posterior predictive pdf evaluated at the `observed_data`. Samples should match with `posterior` ones and each variable should have a counterpart in `observed_data`. The `observed_data` counterpart variable may have a different name.
 
-/observed_data
-    All data that is used in observed variables (or the data (or transformed?) data sections in stan.)
+#### `constant_data`
```

Comment:
I don't really follow this. An example shared between this and `constant_data` would be helpful.

Reply:
I agree: I don't think that the distinction between observed and constant data is clear.  I *suspect* that what is intended is that observed data be "evidence" and constant data be "predictors."  I.e., if we look at the model as a graph, and interpret it causally, the `constant_data` are at the top, and the `observed_data` are at the bottom.

But consider the possibility of a model graph where there are *middle* nodes that are (partially) observed.  (E.g., there is a causal influence that may be observed, but may only be indirectly observed.)  In this case, I believe that the distinction breaks down -- or I am not understanding it right.

Reply:
Yes, it goes with the "Later" comment. What I understand as observed data (which basically comes from how is it currently implemented in ArviZ), all `data` which is used as `pm.<Distribution>(..., observed=data)` in PyMC3 or as `data ~ <distribution>(...)` in Stan goes to observed variable group, as they are modeled as random variables (without taking into account at all if they are then used as parameters of another random variable). Any other relevant data (relevant in the sense of needed to reproduce the analysis) should go to constant data.